dds_web/web/user.py

```diff
@@ -180,9 +180,15 @@ def confirm_2fa():
     if form.validate_on_submit():
         try:
             token = flask.request.cookies["2fa_initiated_token"]
-            user = dds_web.security.auth.verify_token(token)
-        except (KeyError, ddserr.AuthenticationError):
-            flask.flash("You have to first supply valid credentials", "danger")
+            user, _ = dds_web.security.auth.verify_general_token(token)
```

Comment:
this can lead to an attack. As verify_general_token can handle both tokens, an attacker can set the cookie with an encrypted token and extract the data from the return of this function. I think it should not be possible to interpret an encrypted_token from the browser context. Sensitive content that will be stored in the encrypted token can be revealed!